### ISSUE TITLE
yaml strings are quoted unnecessarily

### ISSUE THREAD

**Issue by Benjmhart** [dhall-lang/dhall-haskell#1617]
While building the Bitbucket pipeline typing for dhall - strings were unnecessarily quoted in the resulting yaml file,  which BB does not allow. here is YAML generated:

```yaml
image: "node 10.15.3"
pipelines:
  default:
    - step:
        caches: []
        name: "hello world"
        script:
          - "echo hello"
```

each of the 3 quoted strings ought to be unquoted.

a user-level override would be a good solution here,  maybe an unquoted String type that i could annotate whether my string should be quoted or not when compiling to yaml?

**Comment by Benjmhart** [comment]
here's the branch with code that produced this output: https://github.com/Benjmhart/bitbucket-dhall/tree/yamlbug

**Comment by sjakobi** [comment]
@Benjmhart, can you try `dhall-to-yaml-ng` from the `dhall-yaml` package instead?

@patrickmn, can `aeson-yaml` (which is used in `dhall-to-yaml`) be changed to quote less aggressively?

**Comment by Benjmhart** [comment]
@sjakobi - `dhall-to-yaml-ng` appears to have the correct behaviour here

**Comment by patrickmn** [comment]
Yes, I'll push a change soon that will cause strings like the above not to be quoted.

(aeson-yaml doesn't implement all the escaping rules of yaml. It just quotes anything that MIGHT need to be quoted, e.g. strings including symbols or consisting entirely of numbers. Looks like the rules around spaces are indeed too aggressive -- they just can't be leading spaces.)

**Comment by sjakobi** [comment]
@Benjmhart I'd recommend using `dhall-to-yaml-ng` then, which uses the more mature and fancy `HsYAML` package for YAML rendering.

@patrickmn Thanks! Let's leave this issue open until `aeson-yaml` is fixed and let's add a regression test then.

**Comment by sjakobi** [comment]
@Benjmhart Also, FYI, `dhall-yaml` is licensed with GPL v3, in case that matters to you.

**Comment by Benjmhart** [comment]
@sjakobi - yes I've moved over to `dhall-to-yaml-ng` semi-permanently for the Bitbucket-pipelines dhall project.  thanks for your snappy responses!

**Comment by sjakobi** [comment]
Any news on this, @patrickmn?

**Comment by patrickmn** [comment]
In the latest aeson-yaml master, simple strings containing spaces and dates/bools (https://github.com/dhall-lang/dhall-haskell/issues/1516) are single-quoted. Will cut a release later today and do a PR to re-enable the other test and bump the version.

**Comment by sjakobi** [comment]
@patrickmn If we can fix #1516 that will be great! But this issue is about YAML where no quotes at all are necessary. Have you addressed that too?

```
$ echo '{ x = "hello world" }' | dhall-to-yaml
x: "hello world"
$ echo '{ x = "hello world" }' | dhall-to-yaml-ng
x: hello world
```

**Comment by patrickmn** [comment]
Yes, both issues are addressed.
